packages/webapp/src/pages/Connection/components/SyncRow.tsx

```diff
@@ -7,48 +7,67 @@ import { formatFrequency, getRunTime, parseLatestSyncResult, formatDateToUSForma
 import { Popover, PopoverTrigger } from '../../../components/ui/Popover';
 import { PopoverContent } from '@radix-ui/react-popover';
-import { QuestionMarkCircledIcon } from '@radix-ui/react-icons';
 import { SimpleTooltip } from '../../../components/SimpleTooltip';
-import { IconClockPause, IconClockPlay, IconPlayerPlay, IconRefresh, IconX } from '@tabler/icons-react';
+import { IconClockPause, IconClockPlay, IconRefresh, IconX } from '@tabler/icons-react';
 import { useToast } from '../../../hooks/useToast';
 import { mutate } from 'swr';
+import { Select, SelectItem, SelectContent, SelectTrigger, SelectValue } from '../../../components/ui/Select';
+import { Checkbox } from '../../../components/ui/Checkbox';
+import { apiRunSyncCommand } from '../../../hooks/useSyncs';
 
 export const SyncRow: React.FC<{ sync: SyncResponse; connection: ApiConnectionFull; provider: string | null }> = ({ sync, connection, provider }) => {
     const { toast } = useToast();
 
     const env = useStore((state) => state.env);
-    const runCommandSyncAPI = useRunSyncAPI(env);
 
     const [syncCommandButtonsDisabled, setSyncCommandButtonsDisabled] = useState(false);
 
     const [showPauseStartLoader, setShowPauseStartLoader] = useState(false);
     const [showInterruptLoader, setShowInterruptLoader] = useState(false);
-    const [showTriggerIncrementalLoader, setShowTriggerIncrementalLoader] = useState(false);
+    const [triggerMode, setTriggerMode] = useState<'incremental' | 'full'>(sync.sync_type === 'full' ? 'full' : 'incremental');
+    const [emptyCache, setEmptyCache] = useState(false);
     const [modalSpinner, setModalShowSpinner] = useState(false);
     const [openConfirm, setOpenConfirm] = useState(false);
 
-    const confirmFullRefresh = async () => {
-        if (!sync || syncCommandButtonsDisabled) {
+    const confirmTrigger = async () => {
+        if (!sync || !provider) {
             return;
         }
 
         setSyncCommandButtonsDisabled(true);
         setModalShowSpinner(true);
-        const res = await runCommandSyncAPI('RUN_FULL', sync.schedule_id, sync.nango_connection_id, sync.id, sync.name, provider || '');
+        const res =
+            triggerMode === 'incremental'
+                ? await apiRunSyncCommand(env, {
+                      command: 'RUN',
+                      schedule_id: sync.schedule_id,
+                      nango_connection_id: sync.nango_connection_id,
+                      sync_id: sync.id,
+                      sync_name: sync.name,
+                      provider
+                  })
+                : await apiRunSyncCommand(env, {
+                      command: 'RUN_FULL',
+                      schedule_id: sync.schedule_id,
+                      nango_connection_id: sync.nango_connection_id,
+                      sync_id: sync.id,
+                      sync_name: sync.name,
+                      provider,
+                      empty_cache: emptyCache
+                  });
```

Comment:
This is hard to scan visually and see what's different between the two calls. Would something like this be clearer?

```suggestion
        const res = await apiRunSyncCommand(env, {
                      command: 'RUN',
                      schedule_id: sync.schedule_id,
                      nango_connection_id: sync.nango_connection_id,
                      sync_id: sync.id,
                      sync_name: sync.name,
                      provider,
                      ...(triggerMode === 'incremental' ? {
                        command: 'RUN_FULL',
                        empty_cache: emptyCache,
                      } : {})
                  });
```

Reply:
not a huge fan tbh, both have their own drawbacks